---
title: "Analysing RiboSeq Data"
author: "Alexander Grentner"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Analysing RiboSeq Data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction

This vignette illustratrs how to use the ReactomeGSA R Client with RNA-Seq and Ribo-Seq (or Polysome-Seq) count data, in order to perform a Gene Set Analysis (GSA) over the results of a Differential Translation Analysis (DTA).
A DTA allows to observe to which extent gene expression is driven by transcription or translation, when comparing samples between two conditions. 
The result of such analysis is usually in the form of a table, where each gene is categorised into a Regulatory Mode based on its changes in RNA-Seq derived counts, Ribo-Seq derived counts, or Translational Efficiency (TE) (ie. if it's changes in expression between the two conditions are driven by transcription, translation or both).
The integration with ReactomeGSA allows to further contextualise this information on biological pathways, thus giving an overview on which gene sets are afffected by translation regulation and how.

For additional information regarding Differential Translation Analyses and TE, please refer to [Chotani S. et al. 2019] (https://pubmed.ncbi.nlm.nih.gov/31763789/)

## Installation

The `ReactomeGSA` package can be directly installed from Bioconductor:

```{r, eval=FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE))
  install.packages("BiocManager")

if (!require(ReactomeGSA))
  BiocManager::install("ReactomeGSA")


```

## Getting available methods

The Reactome Analysis System provides multiple pathway algorithms but only PADOG is available for Ribo-Seq data analysis. 
Therefore, to analyse Ribo-Seq data "PADOG" must be selected to choose a modified version for valid Pathway analysis


```{r, echo=FALSE}
library(ReactomeGSA)

my_request <- ReactomeAnalysisRequest(method = "Padog")
my_request
```

All methods for PADOG are available to perform Ribo-Seq analysis. 

```{r, echo=FALSE}
# show the parameter names for the method
padog_params <- available_methods$parameters[available_methods$name == "PADOG"][[1]]
paste0(padog_params$name, " (", padog_params$type, ", ", padog_params$default, ")")
```

## Load data

To perform Ribo-Seq Data analysis a Ribo-Seq file and a RNA-Seq file is needed. Both files need matching column names. Both files are then processed 

```{r, echo=FALSE}

```


## Set parameters

Set parameters for Ribo-Seq analysis request.

```{r, echo=FALSE}
my_request <- set_parameters(request = my_request, max_missing_values = 0.5)
my_request
```





```{r, echo=FALSE}
my_request <- add_dataset(request = my_request, 
                          expression_values = ribo_example, 
                          name = "Ribo Sequencing Data", 
                          type = "ribo_seq",
                          comparison_factor = "condition", 
                          comparison_group_1 = "treated", 
                          comparison_group_2 = "untreated"
                          )
my_request
```

